File: javascript/examples/kappa_numbers.js
var sequences = require('../lib/sequences.js');

let i = 1;
for (let sequenceNumber of sequences.kappaNumberGenerator(9)) {
    process.stdout.write(`k(9, ${(i++).toString().padStart(2)}): ${sequenceNumber.toString().padEnd(30)}`)
    if (i % 3 == 1) {
        console.log();
    }
}

console.log("Using direct computation");
for (let position = 1; position <= 9; ) {
    process.stdout.write(`k(9, ${(position++).toString().padStart(2)}): ${sequences.kappaNumber(9, position).toString().padEnd(30)}`)
    if (position % 3 === 1) {
        console.log();
    }
}

File: dart/examples/kappa_numbers.dart
import 'dart:io';
import 'package:clounce_math/sequences/kappa_numbers.dart';

void main(List<String> args) async {
  num i = 1;
  await for (var sequenceNumber in KappaNumber.generate(9)) {
    stdout.write('k(9, ' + (i++).toString().padLeft(2) + '): ' + sequenceNumber.toString().padRight(30));
    if (i % 3 == 1) {
      print('');
    }
  }

  print('');
  print('Using direct calculation');
  for (var position = 1; position <= 9; position++) {
    stdout.write('k(9, ' + (position).toString().padLeft(2) + '): ' + KappaNumber.compute(9, position).toString().padRight(30));
    if (position % 3 == 0) {
      print('');
    }
  }
}